src/GridFS/StreamWrapper.php

```diff
@@ -54,8 +54,7 @@ class StreamWrapper
     /** @var resource|null Stream context (set by PHP) */
     public $context;
 
-    /** @var ReadableStream|WritableStream|null */
-    private $stream;
+    private ReadableStream|WritableStream|null $stream = null;
```

Comment:
Adding the type on properties produces an error in case of access to uninitialized property. That's why you added the default null value here.

Reply:
That's correct 👍

Reply:
When does PHP raise an error for the uninitialized property? I'm curious if the error is only triggered by explicitly assigning `null` in the property declaration. If so, would you rather do without that and omit `|null` from the type?

Reply:
If the type was only `ReadableStream|WritableStream` without a default value, accessing `$this->stream` would cause a fatal error. I believe we could theoretically use that instead of `assert` in a number of places, as even with disabled assertion handling any access to the stream would result in an object access on `null`, which again yields a fatal error.

I'll note that `isset` does not trigger the uninitialised property error, so we would still be able to check if a stream was initialised. I can change this if y'all prefer to leave `null` out of the type.

Reply:
Feel free to leave as-is. Seems like the path of least resistance.